packages/lsp-compliant-services/src/utils/missingArtifactUtils.ts

```diff
@@ -331,33 +331,124 @@ export class MissingArtifactUtils {
       // Transform LSP position to parser position for symbol manager
       const parserPosition = transformLspToParserPosition(position);
 
+      this.logger.debug(
+        () =>
+          `[extractReferenceAtPosition] LSP position: ${position.line}:${position.character} → ` +
+          `parser position: ${parserPosition.line}:${parserPosition.character} - ` +
+          `uri: ${uri}`,
+      );
+
+      // Get all references in file for debugging
+      const allReferences = (this.symbolManager as any).getAllReferencesInFile
+        ? (this.symbolManager as any).getAllReferencesInFile(uri)
```

Comment:
`as` because it's not in the interface?  why not update that instead of creating permanent type-tech-debt and hiding it with `as`

Reply:
as is, you could remove `getAllReferencesInFile` and the compiler would just smile